src/main/java/org/terasology/volcanoes/SurfaceRasterizer.java

```diff
@@ -40,10 +41,10 @@ public void initialize() {
     @Override
     public void generateChunk(CoreChunk chunk, Region chunkRegion) {
         ElevationFacet elevationFacet = chunkRegion.getFacet(ElevationFacet.class);
-        for (Vector3i position : chunkRegion.getRegion()) {
-            float surfaceHeight = elevationFacet.getWorld(position.x, position.z);
-            if (position.y < surfaceHeight) {
-                chunk.setBlock(ChunkMath.calcRelativeBlockPos(position), stone);
+        for (Vector3ic position : chunkRegion.getRegion()) {
+            float surfaceHeight = elevationFacet.getWorld(position.x(), position.z());
+            if (position.y() < surfaceHeight) {
+                chunk.setBlock(ChunkMath.calcRelativeBlockPos(position, new Vector3i()), stone);

```

Comment:
consider using a `tmp` vector here for reuse.

Reply:
Oh, and I think there's `Chunks.toRelative(position, ...)` as a replacement for `ChunkMath`.